go/libraries/doltcore/sqle/dprocedures/dolt_branch.go

```diff
@@ -164,6 +170,29 @@ func renameBranch(ctx *sql.Context, dbData env.DbData, apr *argparser.ArgParseRe
 		}
 	}
 
+	// If the active branch of the SQL session was renamed, switch to the new branch.
+	if oldBranchName == activeSessionBranch {
```

Comment:
If you implement the method suggested above, then you probably don't need to do any transaction management here at all. The transaction will get committed when this procedure completes anyway

Reply:
You actually want to do this in all cases, regardless of the active branch